zebra/zebra_vrf.h

```diff
@@ -196,6 +201,17 @@ extern struct zebra_vrf *zebra_vrf_lookup_by_name(const char *);
 extern struct zebra_vrf *zebra_vrf_alloc(void);
 extern struct route_table *zebra_vrf_table(afi_t, safi_t, vrf_id_t);
 
+static inline vrf_id_t zebra_vrf_get_evpn_id(void)
+{
+	struct zebra_vrf *zvrf = NULL;
+	zvrf = zebra_vrf_lookup_by_id(VRF_DEFAULT);
```

Comment:
1. It looks like we are storing the vrf_id of the EVPN instance in the default VRF structure. Why should we store EVPN VRF ID in default VRF ?  I am not sure if this is the right place to store this. Shouldn't this be outside the VRF structure ?
2. As per the above change, if default vrf is not present (i.e. zvrf is NULL), then why are we returning VRF_DEFAULT as the vrf id ?

Reply:
> 1. It looks like we are storing the vrf_id of the EVPN instance in the default VRF structure. Why should we store EVPN VRF ID in default VRF ?  I am not sure if this is the right place to store this. Shouldn't this be outside the VRF structure ?

You are right, it is stored in the default VRF. This is because there will always be a default VRF, so we know it will exist. I do agree this is a bit weird, but I am not really aware of anything like `bgp_master` in bgpd. If you (or anyone else) know a better place, please let me know!

> 2. As per the above change, if default vrf is not present (i.e. zvrf is NULL), then why are we returning VRF_DEFAULT as the vrf id ?

Indeed this check is useless as the default VRF always exists. This is because of my habit of always checking pointers, and I can remove the check if you prefer so.

Reply:
I am not concerned about the pointer checks. zvrf will never be NULL, but I think this code conveys a wrong interpretation. i.e. returning "VRF_DEFAULT when zvrf is NULL"

Reply:
I see now. Will definitely remove the check (i.e. straight out return `zvrf->evpn_vrf_id`) in the next revision. Thanks!

Reply:
Done in 77b998f

Reply:
In EVPN use cases, BGP and ZEBRA live together.
the passed vrf_id is present in `bgp->vrf_id` and `zvrf->vrf->vrf_id`.
I don't see the need to have an extra attribute called `evpn_vrf_id`.

can the code be simplified ?
because from vty point of view, evpn underlay will be configured only on one bgp instance; then zvrf->advertise will be confiugred only on one zvrf instance.